src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = [];
+  let newState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(newState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        action.keysToRemove.forEach(key => delete newState[key]);
+        break;
+      case 'clear':
+        newState = {};
+        break;
```

Comment:
It is better to remove all properties from newState object instead of assigning an empty object